static/js/narrow.js

```diff
@@ -776,8 +776,11 @@ exports.deactivate = function () {
       message_list_data structure caching system that happens to have
       home_msg_list in it.
      */
+    coming_from_recent_topics = coming_from_recent_topics || false;
     search.clear_search_form();
-    if (narrow_state.filter() === undefined) {
+    // Both All messages and Recent Topics have `undefiend` filter.
+    // Return if already in the All message narrow.
+    if (narrow_state.filter() === undefined && !coming_from_recent_topics) {
         return;
```

Comment:
I think the `coming_from_recent_topics` check is messy; we should instead check if `current_msg_list === message_list.home`, I think?  I guess I'm not sure what `current_msg_list` should be when we're looking at "Recent topics"; should it be `undefined`?

Reply:
current_msg_list will be `undefined` if user opens czo/#recent_topics first.